api/v1/client/src/commonMain/kotlin/api/RunsApi.kt

```diff
@@ -42,6 +43,33 @@ class RunsApi(
     suspend fun getOrtRun(id: Long): OrtRun =
         client.get("api/v1/runs/$id").body()
 
+    /**
+     * Download the logs of the run with the given [runId] and optional [level] and [steps]. The logs are streamed to
+     * the [streamTarget] function to avoid loading the whole log into memory.
+     * If [level] or [steps] are not provided, the server defaults are used.
+     */
+    suspend fun downloadLogs(
+        runId: Long,
+        level: LogLevel? = null,
+        steps: List<String> = emptyList(),
+        streamTarget: suspend (ByteReadChannel) -> Unit
+    ) {
+        val queryParameters = mutableListOf<String>().apply {
+            level?.let { add("level=${it.name}") }
+            steps.takeIf { it.isNotEmpty() }?.let { add("steps=${it.joinToString(",")}") }
+        }.joinToString("&")
+
+        val url = if (queryParameters.isNotEmpty()) {
```

Comment:
This could be integrated in the `get` function i think? Something like this:

        client.get("api/v1/runs/$runId/logs") {
            url {
                level?.let { parameters.append("level", it.name) }
                steps.joinToString(",").takeIf { it.isNotEmpty() }?.let { parameters.append("steps", it) }
            }
        }

Reply:
Thanks. I like that way more than my approach :)